cockatrice/src/gamescene.cpp

```diff
@@ -44,39 +48,63 @@ void GameScene::addPlayer(Player *player)
 
 void GameScene::removePlayer(Player *player)
 {
-    qDebug("GameScene::removePlayer");
+    qDebug() << "GameScene::removePlayer name=" << player->getName();
     players.removeAt(players.indexOf(player));
     removeItem(player);
     rearrange();
 }
 
+void GameScene::adjustPlayerRotation(int rotationAdjustment)
+{
+    playerRotation += rotationAdjustment;
+    rearrange();
+}
+
 void GameScene::rearrange()
 {
     playersByColumn.clear();
 
+    // Create the list of players playing, noting the first player's index.
     QList<Player *> playersPlaying;
-    int firstPlayer = -1;
-    for (int i = 0; i < players.size(); ++i)
-        if (!players[i]->getConceded()) {
-            playersPlaying.append(players[i]);
-            if ((firstPlayer == -1) && (players[i]->getLocal()))
-                firstPlayer = playersPlaying.size() - 1;
+    int firstPlayerIndex = 0;
+    bool firstPlayerFound = false;
+    QListIterator<Player *> playersIter(players);
+    while (playersIter.hasNext()) {
+        Player *p = playersIter.next();
+        if (!p->getConceded()) {
+            playersPlaying.append(p);
+            if (!firstPlayerFound && (p->getLocal())) {
```

Comment:
Is it not true that there will be either 0 or 1 players where `getLocal() == true`? We might not need the found flag.